zerver/views/realm_export.py

```diff
@@ -68,8 +65,11 @@ def export_realm(request: HttpRequest, user: UserProfile) -> HttpResponse:
             )
         )
 
-    row = RealmAuditLog.objects.create(
-        realm=realm, event_type=event_type, event_time=event_time, acting_user=user
+    row = RealmExport.objects.create(
+        realm=realm,
+        type=RealmExport.EXPORT_PUBLIC,
+        acting_user=user,
+        date_requested=timezone_now(),
     )
```

Comment:
Reading this, I kinda feel like we should have a `PositiveSmallIntegerField` enum state variable, initially `REQUESTED`, so the state machine isn't entirely tracked via these timestamps.

Reply:
Updated with:
```
    REQUESTED = 1
    STARTED = 2
    SUCCEEDED = 3
    FAILED = 4
    DELETED = 5
    status = models.PositiveSmallIntegerField(default=REQUESTED)
```

* Removed the `date_started` field.
* We still need the `date_requested` field for limit check. (5 exports within a week)

---
Question to further simplify the code:

Do you think it's important to differentiate between `STARTED` and `REQUESTED` state ? I think we can simply mark it as `STARTED` when the user makes a request for export (and not wait for concerned worker to actually start the export process). It'll be only helpful if for some reason we want to track those export requests which the worker hasn't yet picked up.

Reply:
Question to further simplify the code:

Do you think it's important to differentiate between `STARTED` and `REQUESTED` state ? I think we can simply mark it as `STARTED` when the user makes a request for export (and not wait for concerned worker to actually start the export process). It'll be only helpful if for some reason we want to track those export requests which the worker hasn't yet picked up.

Reply:
Yes, it is helpful to distinguish for dealing with bugs, in that something in the `STARTED` state we expect to be currently running, while in the `REQUESTED` state we expect it to be in the queue.